cedar-policy-core/src/evaluator/err.rs

```diff
@@ -183,16 +186,23 @@ impl EvaluationError {
     }
 
     /// Construct a [`EntityAttrDoesNotExist`] error
-    pub(crate) fn entity_attr_does_not_exist(
+    pub(crate) fn entity_attr_does_not_exist<'a>(
         entity: Arc<EntityUID>,
         attr: SmolStr,
-        available_attrs: Vec<SmolStr>,
+        available_attrs: impl IntoIterator<Item = &'a SmolStr>,
+        total_attrs: usize,
         source_loc: Option<Loc>,
     ) -> Self {
+        let available_attrs = if total_attrs < TOO_MANY_ATTRS {
+            Some(available_attrs.into_iter().cloned().collect::<Vec<_>>())
+        } else {
+            None
```

Comment:
In the `else` case don't we still want to take the first `TOO_MANY_ATTRS`?

Reply:
We can't, because it's likely coming from a hashmap, so there's no way to deterministically get the same five.

Reply:
Oof. That's unfortunate. I feel like a nondeterministic error message would be a better experience for users than this? but I guess we have other tooling/tests that can't tolerate nondeterministic error messages?

Reply:
I had thought we'd settled on wanting error messages to be deterministic. I agree that in this specific case the non-det results in a more useful error message.

Reply:
I opened #833 based on some annoying non-deterministic in schema parsing errors but I don't think we discussed it. IMO we should keep determinism in error text as a goal,  but it doesn't need to be absolute if it means we can't maintain performance and error message quality.